controllers/auth-controller.js

```diff
@@ -2,44 +2,90 @@ import bcrypt from "bcryptjs"
 import jwt from 'jsonwebtoken';
 import gravatar from 'gravatar';
 import Jimp from 'jimp';
+import { nanoid } from "nanoid";
 import fs from 'fs/promises';
 import path from 'path';
 import User from '../models/user.js'
 
 import { ctrlWrapper } from '../decorators/index.js'
 
-import { HttpError } from '../helpers/index.js'
+import { HttpError, sendEmail, createVerifyEmail } from '../helpers/index.js'
 
 const { JWT_SECRET } = process.env;
 
 const avatarPath = path.resolve("public", "avatars")
 
 const signup = async (req, res) => {
-    console.log('req.body', req.body)
     const { email, password } = req.body;
     const user = await User.findOne({ email });
     if (user) {
-        throw HttpError(409, "Email in use")
+        throw HttpError(409, "Email is been already used")
     }
 
     const avatarURL = gravatar.url(email);
 
     const hashPassword = await bcrypt.hash(password, 10)
 
-    const newUser = await User.create({...req.body, password: hashPassword, avatarURL});
+    const verificationToken = nanoid();
+
+    const newUser = await User.create({ ...req.body, password: hashPassword, verificationToken, avatarURL });
+    
+    const verifyEmail = createVerifyEmail({email, verificationToken})
+
+    await sendEmail(verifyEmail)
+
     res.status(201).json({
         email: newUser.email,
         subscription: newUser.subscription
     })
 }
 
+const verify = async (req, res) => {
+    const { verificationToken } = req.params;
+    const user = await User.findOne({ verificationToken })
+    
+    if (!user) {
+        throw HttpError (404, "Email is not found")
+    }
+
+    await User.findByIdAndUpdate(user._id, { verify: true, verificationToken: "" })
```

Comment:
verificationToken should be null

Reply:
changed to null

Reply:
have you tested it? Will it work correctly?

Reply:
It seems like it is working correctly.
I've attached screens to my homework.